### PR TITLE
Add OOD to slurm.conf

### PR DESCRIPTION
**Pull request by krishmoodbidri**

### CODE REVIEW

**Comment by jprorama** [comment]
There are far  too many stray changes in this PR.  I would expect to see only changes to the slurm.conf template and (likely) the groups_var/all file, defining a conditional variable for the ood nodename inclusion line in template and potentially the name of the ood node itself.

Please make sure you are working off the vagrant-provision branch and update this PR.  It looks like you are working off of my master which is the upstream master and not vagrant-provision.

**Comment by jprorama** [comment]
Please remove the test commits from this PR.  If you need to test a idea, please use a test repo.  Likewise, the cluster-root.pub file should not be part of this commit.

Please add a description to your pull request summarizing the motivation for the PR.

File: group_vars/all [hunk 1]
```diff
@@ -66,6 +66,10 @@
   login_node_prefix: "login-"
   num_login_nodes: 0
 
+#OpenOnDemand
+  ood: "ood"
+  ood_provision: true
+
```
Comment (krishmoodbidri) [inline group_vars/all]: Got a warning while running 
ansible-playbook -c local -i /vagrant/CRI_XCBC/hosts -l `hostname` /vagrant/CRI_XCBC/ohpc.yaml --tags=pre_ohpc -s -vv

[WARNING]: Found both group and host with same name: ood

File: roles/ohpc_install/templates/slurm_conf.j2 [hunk 2]
```diff
@@ -108,3 +108,8 @@ NodeName=c0 Sockets=1 CoresPerSocket=1 State=UNKNOWN
 #PartitionName=high Nodes=compute-[0-1] Default=YES MaxTime=INFINITE State=UP PriorityTier=10
 #PartitionName=gpu  Nodes=gpu-compute-1 Default=YES MaxTime=INFINITE State=UP PriorityTier=5 AllowGroups=slurmusers
 PartitionName=low  Nodes=c0 Default=YES MaxTime=2-00:00:00 State=UP
+
+#OOD
+{% if ood_provision %}
+	NodeName=={{ ood }}
```
Comment (jprorama) [inline roles/ohpc_install/templates/slurm_conf.j2]: Check and make sure this leading white space on the NodeName is correct syntax.   I suggest removing it regardless.  Keep in mind that we aren't doing 'good coding block indent patterns' with these templates.  They are literally just inserting text into a config file.  Unless the white space makes sense in the resulting config file, it likely shouldn't be in the template.

File: group_vars/all [hunk 3]
```diff
@@ -66,6 +66,10 @@
   login_node_prefix: "login-"
   num_login_nodes: 0
 
+#OpenOnDemand
+  ood: "ood"
```
Comment (jprorama) [inline group_vars/all]: Let's change this variable to ood_nodename.  It better describes its function and avoids confusion between a variable named ood and a value of ood.

File: roles/ohpc_install/templates/slurm_conf.j2 [hunk 4]
```diff
@@ -108,3 +108,8 @@ NodeName=c0 Sockets=1 CoresPerSocket=1 State=UNKNOWN
 #PartitionName=high Nodes=compute-[0-1] Default=YES MaxTime=INFINITE State=UP PriorityTier=10
 #PartitionName=gpu  Nodes=gpu-compute-1 Default=YES MaxTime=INFINITE State=UP PriorityTier=5 AllowGroups=slurmusers
 PartitionName=low  Nodes=c0 Default=YES MaxTime=2-00:00:00 State=UP
+
+#OOD
+{% if ood_provision %}
+NodeName=={{ ood }}
```
Comment (jprorama) [inline roles/ohpc_install/templates/slurm_conf.j2]: Naturally, this variable ref will need to be updated to align with the var change in groups_var/all

**Comment by jprorama** [comment]
Note: the early comments in this pull request refer to an earlier state of the source repo branch that was rebased, dropping the commits related to "tests".  The branch was updated via a force push erasing that commit history.

**Pull request merged**